core/src/main/java/org/elasticsearch/search/aggregations/InternalAggregation.java

```diff
@@ -140,6 +135,11 @@ public String getName() {
      */
     public abstract Type type();
 
+    @Override
+    public String getWriteableName() {
+        return type().stream().toUtf8();
```

Comment:
It is *correct* in that it works. I think we should stop doing it but in another PR. Or I can do it in this one, I guess, but it'd make it larger.